backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts

```diff
@@ -0,0 +1,57 @@
+import { type SendMailOptions } from 'nodemailer';
+
+import { EmailPayload } from '../constants/email-payload.js';
+import { EmailSubject } from '../enums/email-subject.enum.js';
+import { type SubscriptionDetails } from '../types/types';
+
+const MILLIS_TO_SECOND = 1000;
+
+const loadAndReplaceTemplate = (variables: Record<string, string>): string => {
+    let htmlContent = EmailPayload.HTML;
+    for (const key of Object.keys(variables)) {
+        const regex = new RegExp(`{${key}}`, 'g');
+        htmlContent = htmlContent.replace(regex, variables[key]);
+    }
+
+    return htmlContent;
+};
```

Comment:
```suggestion
const loadAndReplaceTemplate = (variables: Record<string, string>): string => {
    let htmlContent = EmailPayload.HTML;
    for (const key of Object.keys(variables)) {
        const regex = new RegExp(`{${key}}`, 'g');
        htmlContent = htmlContent.replace(regex, variables[key]);
    }

    return htmlContent;
};
```

```suggestion
  let htmlContent = EmailPayload.HTML;
    for (const [key, value] of Object.entries(variables)) {
        htmlContent = htmlContent.split(`{${key}}`).join(value);
    }
    return htmlContent;
```

regex can be avoided here